pkg/sloop/webserver/debug.go

```diff
@@ -123,23 +124,46 @@ func listKeysHandler(tables typed.Tables) http.HandlerFunc {
 		var keys []string
 
 		count := 0
+		totalCount := 0
+		var totalSize int64 = 0
 		err = tables.Db().View(func(txn badgerwrap.Txn) error {
-			keyPrefix := "/" + table + "/"
+			keyPrefix := ""
+			if table != "internal" {
+				keyPrefix = "/" + table + "/"
+			}
+
 			iterOpt := badger.DefaultIteratorOptions
 			iterOpt.Prefix = []byte(keyPrefix)
+			iterOpt.AllVersions = true
+			iterOpt.InternalAccess = true
 			itr := txn.NewIterator(iterOpt)
 			defer itr.Close()
 
-			for itr.Seek([]byte(keyPrefix)); itr.ValidForPrefix([]byte(keyPrefix)); itr.Next() {
+			for itr.Rewind(); itr.Valid(); itr.Next() {
```

Comment:
unfamiliar with the subtleties of badger - is there much of a difference between previous loop construct and new one? I was initially wondering if the loop is going to be looking at a lot more rows then before.

Reply:
There is a difference that it actually loops over the keys instead of seek which is relatively a faster call. Even though it should work as expected, however unfortunately in the current setup the deleted keys do end up showing in seek where I had spend a lot of time to debug.

For now I am keeping Rewind instead of seek. However it is one of the things in my todo to see why seek is not behaving appropriately.

On the other hand, thanks to your comment, I just noticed that I am not calling ValidForPrefix which is still required. I will update that.